### ISSUE TITLE
Data parsing Bug for dataloader.

### ISSUE THREAD

**Issue by panganqi** [aqlaboratory/openfold#17]
Hi, I've processed some data for training but get the bug of dataloader of:

`File "/share/home/openfold/openfold-main/lib/conda/envs/openfold_venv/lib/python3.7/site-packages/pytorch_lightning/utilities/apply_func.py", line 92, in apply_to_collection
    return function(data, *args, **kwargs)
  File "/share/home/openfold/openfold-main/lib/conda/envs/openfold_venv/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 521, in __next__
    data = self._next_data()
  File "/share/home/openfold/openfold-main/lib/conda/envs/openfold_venv/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 1203, in _next_data
    return self._process_data(data)
  File "/share/home/openfold/openfold-main/lib/conda/envs/openfold_venv/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 1229, in _process_data
    data.reraise()
  File "/share/home/openfold/openfold-main/lib/conda/envs/openfold_venv/lib/python3.7/site-packages/torch/_utils.py", line 434, in reraise
    raise exception
TypeError: Caught TypeError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/share/home/openfold/openfold-main/lib/conda/envs/openfold_venv/lib/python3.7/site-packages/torch/utils/data/_utils/worker.py", line 287, in _worker_loop
    data = fetcher.fetch(index)
  File "/share/home/openfold/openfold-main/lib/conda/envs/openfold_venv/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py", line 52, in fetch
    return self.collate_fn(data)
  File "/share/home/openfold/openfold-main/lib/conda/envs/openfold_venv/lib/python3.7/site-packages/pytorch_lightning/utilities/auto_restart.py", line 474, in _capture_metadata_collate
    data = default_collate(samples)
  File "/share/home/openfold/openfold-main/openfold/data/data_modules.py", line 297, in __call__
    prot, self.stage
  File "/share/home/openfold/openfold-main/openfold/data/feature_pipeline.py", line 116, in process_features
    mode=mode,
  File "/share/home/openfold/openfold-main/openfold/data/feature_pipeline.py", line 93, in np_example_to_features
    cfg[mode],
  File "/share/home/openfold/openfold-main/openfold/data/input_pipeline.py", line 187, in process_tensors_from_config
    lambda x: wrap_ensemble_fn(tensors, x), torch.arange(num_recycling + 1)
  File "/share/home/openfold/openfold-main/openfold/data/input_pipeline.py", line 201, in map_fn
    ensembles = [fun(elem) for elem in x]
  File "/share/home/openfold/openfold-main/openfold/data/input_pipeline.py", line 201, in <listcomp>
    ensembles = [fun(elem) for elem in x]
  File "/share/home/openfold/openfold-main/openfold/data/input_pipeline.py", line 187, in <lambda>
    lambda x: wrap_ensemble_fn(tensors, x), torch.arange(num_recycling + 1)
  File "/share/home/openfold/openfold-main/openfold/data/input_pipeline.py", line 168, in wrap_ensemble_fn
    return fn(d)
  File "/share/home/openfold/openfold-main/openfold/data/data_transforms.py", line 76, in <lambda>
    return lambda x: f(x, *args, **kwargs)
  File "/share/home/openfold/openfold-main/openfold/data/input_pipeline.py", line 196, in compose
    x = f(x)
  File "/share/home/openfold/openfold-main/openfold/data/data_transforms.py", line 76, in <lambda>
    return lambda x: f(x, *args, **kwargs)
  File "/share/home/openfold/openfold-main/openfold/data/data_transforms.py", line 180, in sample_msa
    num_seq = protein["msa"].shape[0]
TypeError: 'function' object is not subscriptable`

And I upload one of the datasample, Am I wrong with the generate MSAs pipeline or wrong with the dataloader?
[5E0Y.zip](https://github.com/aqlaboratory/openfold/files/7582029/5E0Y.zip)

**Comment by gahdritz** [comment]
Could you share the command you're running?

**Comment by panganqi** [comment]
```python scripts/data_dir_to_fasta.py mmcif_data processed_fasta/input.fasta`
`python3 scripts/precompute_alignments_mmseqs.py processed_fasta/input.fasta data/mmseqs_dbs uniref30_2103_db alignment_dir_single/ ./mmseqs/bin/mmseqs --hhsearch_binary_path=lib/conda/envs/openfold_venv/bin/hhsearch --env_db colabfold_envdb_202108_db --pdb70 data/pdb70/pdb70`

Here I get the 5E0Y_1_A_A folder under alignment_dir_single folder, which contains the a3m hhr files.

`python3 scripts/generate_mmcif_cache.py mmcif_data/ mmcif_cache.json --no_workers 16`

get the mmcif_cache.json( I change the release_data):
`{
    "5e0y_1_A": {
        "release_date": "2017-03-29",
        "no_chains": 1
    }
}`


Then I run
`python3 train_openfold.py mmcif_data/ alignment_dir_single/ data/pdb_mmcif/mmcif_files/ output 2021-10-10 --template_release_dates_cache_path mmcif_cache.json --precision 16 --gpus 0 --seed 42`

To test dataloader. And get the bug.

**Comment by gahdritz** [comment]
Fixed in 79f9f03. It ended up being a bug in the distillation set pipeline.

Two notes about your usage:

1. Note that --gpus 0 means "use 0 GPUs", not "use GPU 0". To limit the model to a specific GPU, use the `CUDA_VISIBLE_DEVICES` environment variable.
2. `.pdb` files in the `mmcif_dir` are interpreted as examples from the distillation set and are treated slightly differently from ground truth examples. You'll want to get rid of it if this is a real protein. This wasn't at all clear in the previous README, so I added a paragraph about that too.
